driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -253,9 +253,11 @@ public int getConnectTimeoutMs() {
                 () -> throwMongoTimeoutException("The operation exceeded the timeout limit.")));
     }
 
-    public void resetTimeout() {
-        assertNotNull(timeout);
-        timeout = startTimeout(timeoutSettings.getTimeoutMS());
+    public void resetTimeoutIfPresent() {
+        if (hasTimeoutMS()) {
```

Comment:
Comment: The method naming makes the `assertNotNull` seem restrictive. Should it be part of the if statement?

I guess its a bug if this is called if there isnt a timeout set - its just slightly unclear thats the case.